File: LearningPlan.DomainModel/Plan.cs
using System;

namespace LearningPlan.DomainModel
{
    public class Plan : EntityBase
    {
        public Plan(string name, string userId)
        {
            Name = name;
            UserId = userId;
        }
        public string Name { get; set; }
        public string UserId { get; set; }
        public bool IsTemplate { get; set; }
    }
}

File: LearningPlan.Web.v2/src/models/plan.ts
import type { PlanArea } from "./plan-area"

export class Plan {
  id: string
  name: string
  planAreas: PlanArea[] = []
  isTemplate: boolean = false
  userId: string = ''
  constructor(id: string, name: string) {
    this.id = id
    this.name = name
  }
}
